docs/getting-started/data.md

```diff
@@ -0,0 +1,1202 @@
+# Gutenberg data layer
+
+## Pre-requisites
+
+This doc assumes you are familiar with Redux concepts like action, reducer, and selector, as well as certain Gutenberg concepts like _Thunks_. You may still be able to get the gist of the ideas here without these pre-requisites, but you are highly encouraged to get familiar with them first.
+
+## Big Ideas
+
+Gutenberg is part of WordPress core and frequently acts on the same data. There are posts, pages, taxonomies, widgets, navigation items and so on. The obvious way of using the data, would be to just request it from API whenever a given React component needs it. This would have a serious drawback, though. Any other component wanting to use the same data, would have to request it too. So now there is more than one request. Then, what if the data changes? Would all components re-request it? Would they even know?
+
+The data layer provides answers to all of these questions, and more. It handles data synchronization for you, so you can focus on your component. For example, when you need to do something with a post, you can access it with:
+
+```js
+function MyComponent({ widgetId }) {
```

Comment:
This seemed strange to me when I read it. You referenced the term `post` but then this example is specific to Widgets.